python/dgllife/utils/featurizers.py

```diff
@@ -2081,3 +2155,171 @@ def __init__(self, bond_data_field='e', self_loop=False):
+    self loops.**
+
+    Parameters
+    ----------
+    bond_data_field : str
+        Name for storing bond features in DGLGraphs, default to ``'e'``.
+    max_length : int
+        Maximum distance upto to which shortest paths must be considered.
+        Paths shorter than max_length will be padded and longer will be
+        truncated, default to ``5``.
+
+    Examples
+    --------
+
+    >>> from dgllife.utils import PAGTN_EdgeFeaturizer
+    >>> from rdkit import Chem
+    >>> mol = Chem.MolFromSmiles('CCO')
+    >>> bond_featurizer = PAGTN_EdgeFeaturizer(max_length= 1)
+    >>> bond_featurizer(mol)
+    {'e': tensor([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]])}
+    >>> # Get feature size
+    >>> bond_featurizer.feat_size()
+    14
+
+    See Also
+    --------
+    BaseBondFeaturizer
+    CanonicalBondFeaturizer
+    WeaveEdgeFeaturizer
+    PretrainBondFeaturizer
+    """
+    def __init__(self, bond_data_field='e', max_length=5):
+        self.bond_data_field = bond_data_field
+        self.RING_TYPES = [(5, False), (5, True), (6, False), (6, True)]
+        self.ordered_pair = lambda a, b: (a, b) if a < b else (b, a)
+        self.bond_featurizer = ConcatFeaturizer([bond_type_one_hot,
+                                                 bond_is_conjugated,
+                                                 bond_is_in_ring])
+        self.max_length = max_length
+
+    def feat_size(self):
+        """Get the feature size.
+
+        Returns
+        -------
+        int
+            Feature size.
+        """
+        mol = Chem.MolFromSmiles('C')
+        feats = self(mol)[self.bond_data_field]
+
+        return feats.shape[-1]
+
+    def bond_features(self, mol, path_atoms, ring_info):
+        """Computes the edge features for a given pair of nodes.
+
+        Parameters
+        ----------
+        mol : rdkit.Chem.rdchem.Mol
+            RDKit molecule instance.
+        path_atoms: list
+            Shortest path between the given pair of nodes.
+        ring_info: list
+            Diffrent rings by which they are connected
+        """
+        features = []
+        path_bonds = []
+        path_length = len(path_atoms)
+        for path_idx in range(len(path_atoms) - 1):
+            bond = mol.GetBondBetweenAtoms(path_atoms[path_idx], path_atoms[path_idx + 1])
+            assert bond is not None
```

Comment:
We should probably just skip it with a warning rather than interrupt the program.